src/app/components/module-index-page/module-index-page.component.ts

```diff
@@ -1,193 +1,389 @@
+        );
+
+        this.normalizePriority(module);
+    }
+
+    /**
+     * Lists the content for module input or complete closes the node
+     * @param module - Module to check for expansion
+     * @param event  - Used to filter out double requests since this is a parent element event
+     */
+    listOrCloseContent(module: Module, event) {
+
+        if (event) {
+
+            const target = event.target || event.srcElement || event.currentTarget;
+            const idAttr: string = target.attributes.id && (target.attributes.id.nodeValue || '');
+
+            if (idAttr && !idAttr.includes('mat-expansion-panel')) {
+
+                return;
+            }
+        }
+
+        if (this.contentVisible.get(module) || this.childrenVisible.get(module)) {
+
+            this.contentVisible.set(module, false);
+            this.childrenVisible.set(module, false);
+            return;
+        }
+
+        this.listContent(module);
+    }
+
+    /**
+     * Expands the children display option for the Node
+     * @param module - Module to expand
+     */
+    listChildren(module: Module) {
+
+        for (const key of this.contentVisible.keys()) {
+
+            if (key.id !== module.id) {
+
+                this.contentVisible.set(key, false);
+                this.childrenVisible.set(key, false);
+            }
+        }
+
+        this.childrenVisible.set(module, !this.childrenVisible.get(module));
+        this.contentVisible.set(module, false);
+        this.setActiveChild(module, 0);
+        this.normalizePriority(this.activeModule);
+    }
+
+    /**
+     * Send a request to remove the link between the selected Link and Module
+     * If the response is a success, then we will delete the module link locally
+     */
+    removeContentFromModuleIndex() {
+
+        globalCacheBusterNotifier.next();
+
+        this.cs.removeLinkFromContent(this.selLink.id).subscribe(
+
+            (resp) => {
+
+                this.selModule.links.splice(this.contentActive.get(this.selModule), 1);
+                this.contentActive.set(this.selModule, this.selModule.links.length === 0 ? -1 : 0);
+                this.normalizePriority(this.selModule);
+            }
+        );
+    }
+    /**
+     * Assigns the link and module that will be used for link removal
+     * @param link - the selected link
+     * @param module - the module the selected link resides in
+     */
+    selectedLinkForRemoval(link: Link, module: Module) {
+
+        this.selLink = link;
+        this.selModule = module;
+    }
+
+    /**
+     * Assigns the module that will be used for module removal
+     * @param module - module that will be writen as selected
+     */
+    selectedModuleForRemoval(module: Module) {
+        this.selModule = module;
+    }
+
+    /**
+     * Sends a request to delete the module selected
+     */
+    removeModule() {
+
+        const selMethod = (document.getElementById('Seldelmethod') as HTMLInputElement).value;
```

Comment:
Why use the document object here rather than databinding?

Reply:
Artifact of last batch that worked on it. Removed and attached ngModel in the HTML